src/utils/arrayUtils.ts

```diff
@@ -14,3 +14,14 @@ export function chunk<T>(array: T[], chunkSize: number): T[][] {
   // eslint-disable-next-line @typescript-eslint/no-unsafe-return
```

Comment:
🟥 Feel free to ignore this comment as beyond the scope of this PR... but I really don't like what's going on in the `chunk` function. Ignoring `no-unsafe-return` is dangerous, and the implementation is especially fiddly and prone to error. 

Chunking can be expressed rather naturally with a generator function which yields one chunk at a time. We can gather them into an array with `Array.from`. Here's an example alternative implementation:

```ts
function* chunkGenerator<Item>(
  array: readonly Item[],
  chunkSize = 1,
) {
  let index = 0;

  while (index < array.length) {
    yield array.slice(index, index + chunkSize);
    index += chunkSize;
  }
}

const chunk = <Item>(
  array: readonly Item[],
  chunkSize = 1,
) => Array.from(chunkGenerator(array, chunkSize));
```

Reply:
That's fair. My instinct had been "this is something I feel very strongly about"... but it is naturally contradicted by "... but it might not at all be relevant to this PR". 

I'll edit to 🟢